podpac/core/interpolation/interpolation.py

```diff
@@ -491,14 +495,26 @@ def interpolate(self, source_coordinates, source_data, eval_coordinates, output_
         self._last_interpolator_queue = interpolator_queue
 
         # iterate through each dim tuple in the queue
-        for udims in interpolator_queue:
-            interpolator = interpolator_queue[udims]
-
-            # run interpolation
-            output_data = interpolator.interpolate(
-                udims, source_coordinates, source_data, eval_coordinates, output_data
+        outputs = output_data.coords.get("output")
+        dtype = output_data.dtype
+        for udims, interpolator in interpolator_queue.items():
+            # TODO use short-circuits within this loop
+
+            # interp_coordinates are essentially intermediate eval_coordinates
+            interp_dims = [dim for dim, c in source_coordinates.items() if set(c.dims).issubset(udims)]
+            other_dims = [dim for dim, c in eval_coordinates.items() if not set(c.dims).issubset(udims)]
+            interp_coordinates = merge_dims([source_coordinates.drop(interp_dims), eval_coordinates.drop(other_dims)])
+            interp_data = UnitsDataArray.create(interp_coordinates, outputs=outputs, dtype=dtype)
+            interp_data = interpolator.interpolate(
+                udims, source_coordinates, source_data, interp_coordinates, interp_data
             )
 
+            # prepare for the next iteration
+            source_data = interp_data
+            source_coordinates = interp_coordinates
+
+        output_data.data = interp_data.transpose(*output_data.dims)
+
```

Comment:
@mlshapiro I really only made two essential changes to support heterogeneous interpolation. This one creates intermediate `interp_coordinates` and `interp_data`. Basically the data starts out as `source_data` and then becomes more like `output_data` with interation.

Reply:
this seems fine to me as long as it doesn't present any strange pass-by-reference issues. Do you need to copy interp_data or anything like that to make sure that it doesn't get mutated through the iterations? YOu are a better resource for this  kind of python-specific question, so I trust you did this right.

Reply:
Yeah, it's fine because `interp_data` is fully (re)assigned on line 507.